File: SafetyAI/src/providers/anthropic.ts
import { request } from 'undici';
import { LLMClient, LLMResponse } from './index.js';

export class AnthropicClient implements LLMClient {
  constructor(private apiKey: string) {}
  name() { return 'anthropic'; }

  async classify(opts: { model: string; system: string; user: string }): Promise<LLMResponse> {
    const url = 'https://api.anthropic.com/v1/messages';
    const body = {
      model: opts.model,
      system: opts.system,
      max_tokens: 256,
      temperature: 0,
      messages: [
        { role: 'user', content: opts.user }
      ]
    };
    const res = await request(url, {
      method: 'POST',
      body: JSON.stringify(body),
      headers: {
        'content-type': 'application/json',
        'x-api-key': this.apiKey,
        'anthropic-version': '2023-06-01'
      }
    });
    const data = await res.body.json() as any;
    const content = data?.content?.[0]?.text ?? '';
    let json: any | undefined;
    try { json = JSON.parse(content); } catch {}
    return { raw: content, json };
  }
}

File: apps/webserverGo/safety/providers/anthropic.go
package providers

import (
	"bytes"
	"encoding/json"
	"io"
	"net/http"
)

type anthropicClient struct {
	apiKey string
}

func NewAnthropicClient(apiKey string) Client {
	return &anthropicClient{apiKey: apiKey}
}

func (c *anthropicClient) Name() string { return "anthropic" }

func (c *anthropicClient) Classify(model, systemPrompt, userPrompt string) (Response, error) {
	body := map[string]any{
		"model": model,
		"system": systemPrompt,
		"max_tokens": 256,
		"temperature": 0,
		"messages": []map[string]any{{"role": "user", "content": userPrompt}},
	}
	bts, _ := json.Marshal(body)
	req, _ := http.NewRequest(http.MethodPost, "https://api.anthropic.com/v1/messages", bytes.NewReader(bts))
	req.Header.Set("content-type", "application/json")
	req.Header.Set("x-api-key", c.apiKey)
	req.Header.Set("anthropic-version", "2023-06-01")

	resp, err := http.DefaultClient.Do(req)
	if err != nil {
		return Response{}, err
	}
	defer resp.Body.Close()
	data, _ := io.ReadAll(resp.Body)

	var res struct {
		Content []struct{
			Text string `json:"text"`
		} `json:"content"`
	}
	_ = json.Unmarshal(data, &res)
	content := ""
	if len(res.Content) > 0 {
		content = res.Content[0].Text
	}
	var js map[string]any
	_ = json.Unmarshal([]byte(content), &js)
	return Response{Raw: content, JSON: js}, nil
}
